sdks/go/test/integration/io/fhirio/fhirio_test.go

```diff
@@ -0,0 +1,198 @@
+// detrimental to the tests, so it is fine to ignore.
+func populateStore(storePath string) []string {
+	resourcePaths := make([]string, 0)
+	for _, bundle := range readPrettyBundles() {
+		response, err := storeService.ExecuteBundle(storePath, bytes.NewReader(bundle)).Do()
+		if err != nil {
+			continue
+		}
+
+		var body struct {
+			Entry []struct {
+				Response struct {
+					Location string `json:"location"`
+					Status   string `json:"status"`
+				} `json:"response"`
+			} `json:"entry"`
+		}
+		err = json.NewDecoder(response.Body).Decode(&body)
+		if err != nil {
+			continue
+		}
+
+		for _, entry := range body.Entry {
+			bundleFailedToBeCreated := !strings.Contains(entry.Response.Status, "201")
+			if bundleFailedToBeCreated {
+				continue
+			}
+
+			resourcePath, err := extractResourcePathFrom(entry.Response.Location)
+			if err != nil {
+				continue
+			}
+			resourcePaths = append(resourcePaths, resourcePath)
+		}
+	}
+	return resourcePaths
+}
+
+func readPrettyBundles() [][]byte {
+	files, _ := os.ReadDir(testDataDir)
+	bundles := make([][]byte, len(files))
+	for i, file := range files {
+		bundles[i], _ = os.ReadFile(testDataDir + file.Name())
+	}
+	return bundles
+}
+
+func extractResourcePathFrom(resourceLocationUrl string) (string, error) {
+	// The resource location url is in the following format:
+	// https://healthcare.googleapis.com/v1/projects/PROJECT_ID/locations/LOCATION/datasets/DATASET_ID/fhirStores/STORE_ID/fhir/RESOURCE_NAME/RESOURCE_ID/_history/HISTORY_ID
+	// But the API calls use this format: projects/PROJECT_ID/locations/LOCATION/datasets/DATASET_ID/fhirStores/STORE_ID/fhir/RESOURCE_NAME/RESOURCE_ID
+	startIdx := strings.Index(resourceLocationUrl, "projects/")
+	endIdx := strings.Index(resourceLocationUrl, "/_history")
+	if startIdx == -1 || endIdx == -1 {
+		return "", errors.New("resource location url is invalid")
+	}
+	return resourceLocationUrl[startIdx:endIdx], nil
+}
+
+func TestFhirIO_Read(t *testing.T) {
+	integration.CheckFilters(t)
+	checkFlags(t)
+
+	_, testResourcePaths, teardownFhirStore := setupFhirStore(t)
+	defer teardownFhirStore()
+
+	p := ReadPipeline(testResourcePaths)
+	ptest.RunAndValidate(t, p)
+}
+
+func TestFhirIO_InvalidRead(t *testing.T) {
+	integration.CheckFilters(t)
+	checkFlags(t)
+
+	fhirStorePath, _, teardownFhirStore := setupFhirStore(t)
+	defer teardownFhirStore()
+
+	p := InvalidReadPipeline(fhirStorePath)
+	ptest.RunAndValidate(t, p)
+}
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	beam.Init()
+
+	healthcareService, err := healthcare.NewService(context.Background(), option.WithUserAgent(fhirio.UserAgent))
+	if err == nil {
+		storeService = healthcare.NewProjectsLocationsDatasetsFhirStoresFhirService(healthcareService)
+		storeManagementService = healthcare.NewProjectsLocationsDatasetsFhirStoresService(healthcareService)
+	}
+
+	beam.PipelineOptions.LoadOptionsFromFlags(nil)
+	gcpProject = beam.PipelineOptions.Get("project")
```

Comment:
I assume this bit is for checkFlags to work properly? I'm worried there might be unintended issues with calling these methods manually here. Instead could you just check the Project and Region variables from the [gcpopts](https://github.com/apache/beam/blob/master/sdks/go/pkg/beam/options/gcpopts/options.go) package directly wherever you need it? That seems like it would work better.